tests/test-dump-counters/test-dump-counters.py

```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env python3
+import argparse
+import json
+import logging
+import os
+import os.path
+import signal
+import sys
+import time
+
+import suricatasc
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--pidfile")
+    parser.add_argument("--suricata-socket")
+    parser.add_argument("--poll-interval", type=float, default=0.1)
+    parser.add_argument("--timeout", type=float, default=5)
+    parser.add_argument("--verbose", default=False, action="store_true")
+    args = parser.parse_args()
+
+    logging.basicConfig(level=logging.DEBUG if args.verbose else logging.INFO)
+
+    poll_interval = args.poll_interval
+    end_time = time.time() + args.timeout
+    suricata_pid = None
+    exit_code = 1
+
+    while time.time() < end_time:
```

Comment:
in a private bash script unix test I loop until the `version` unix command succeeds